### ISSUE TITLE
Rationalise use cases of alert banner and service update / status

### ISSUE THREAD

**Issue by andybroomfield** [localgovdrupal/localgov#255]
Two PRs will give Alert banner localgovdrupal/localgov_alert_banner#142 and service status localgovdrupal/localgov_services#115 some new powers, allowing alerts to be targeted to specific pages.

Its not clear what the difference between these approachs are, or why one should be used over the other.
I'm a little concered about developing franken features and it would help if the use cases where documented on why we have two approaches to putting an alert on a page.

**Comment by willguv** [comment]
I’ve started at chat with Nikki in Brighton, which I think will lead to useful rationalisation

We were talking about the existing BHCC service landing page which has a “service updates” panel beneath the fold:

> We know that users take different routes to service info pages and if they go through search or follow a link from elsewhere on the site and are likely to bypass the service hub – even if they go through there they might miss it at the bottom of the page.
>  
> For this reason, particularly with Covid information, services have often asked us to duplicate service notices at the top of several pages- which makes it a bigger job and more likely to miss some when we need to remove them or make changes. Perhaps we could have an option to apply an update to all pages under a service or sub hub?

**Comment by andybroomfield** [comment]
Hope you don't mind I edited your comment @willguv to change to pre text to a blockqoute.

I agree with what is being expressed, I'm concered that service status is becoming a mega feature. 
For alerts, Alert banner strikes me as the better place, keeping all alerts in a central place and developing an in page alert type that can be placed below the title and be visible on any page that is desiered (not just those in the services stack), and the link can be any link. 

Counter point, having the in page alert as part of service status does mean a seperate alert does not have to be created so it can be done as part of adding a service status / update.

**Comment by willguv** [comment]
@andybroomfield It doesn't feel about a mega feature to me. As a content designer, I'll be able to write a status update message about, say, bin collections and be able to display it on either:

- The service status page at https://demo.localgovdrupal.org/service-status
- A curated selection of other pages on the site as needed

Happy to chat about this some more

**Comment by andybroomfield** [comment]
@willguv  What I mean is that service status feels like three seperate features.
1. The service updates nodes that exist under the landing page (as on City Clean)
2. The service status page which is a way of saying how functional different services are
3. In page alerts on service pages that link back to the service status.

For the purpose of this issue, we need to decide if the in page alerts (3) crosses over with functionality provided by the alert banner, and if it would be better if Alert banner was the caninocial way of adding any in page alerts. We could create a new alert banner type (in page alert) and style appropriatly.

... Or, counterpoint, we could decide that actully being able to create these at the same time as writing a service status is better for content desingers as they don't need to write it again in another place.

Basically, I'm looking to get a sense of what each of these features is for and its intended use.

**Comment by willguv** [comment]
@andybroomfield thanks for the info.

It seems to me that 3) and 1) are related needs:

- content designers need to add service status messages to selected pages
- councils want to show all current service status messages on a single page (this has been critical during lockdown, not sure about afterwards)

It feels like service updates on landing pages 1) isn't wanted by anyone and should be removed. It was an untested idea at the time - a design that was relatively cheap to implement so worth a try.

Croydon don't think it's fit for purpose, and BHCC want to put messages on many pages. I'll know more when I talk to Nikki next week.

**Comment by andybroomfield** [comment]
There is definitely a need to display a service status alert message on selected pages inside the service stack.

For the purpose of this issue, the question is which module should be responsible for it, do people think service status module (do it when you write a service update) or should alert banner (do all the alerts in one place). would be the best place.

**Comment by andybroomfield** [comment]
@willguv Planning on BHCC is using service updates recently.
https://www.brighton-hove.gov.uk/content/planning

**Comment by willguv** [comment]
Thanks for the examples @andybroomfield 

Do you know if these were created by Chloe or Nikki, or the planning team?

Many thanks
